### DISCUSSION TITLE
Mib2 bootloop

### DISCUSSION THREAD

**Discussion by KJW13** [Mr-MIBonk/M.I.B._More-Incredible-Bash#181] [Q&A]
hello
I have a problem with my MIB2 High device.
Upgrading to MHI2_ER_SKG13_P4526 (MU: 1440) was successful.
I cleared the software installation error.
In the first installation attempt after inserting the SD card with the latest M.I.B and IFS-stage2 build, the device started the update automatically and completed it successfully. In the service mode, the green menu appeared, but there was no M.I.B tab.
I cleared the software installation error.
I made the second installation attempt through the update installation options. After successful completion, the M.I.B tab appeared in the green menu. I made a large backup.
The next day the device greeted me with a bootloop. Logo, sometimes menu view and reboot after 60s
Using D-Link USB-Ethernet and PuTTY in RCC mode, it does not connect to the device.
In MMX mode, I can login but "rm /mnt/efs-persist/FEC/mmx_fec_ids.bin" returns a missing file or folder error.
Is there a chance to save the device? What else can I do myself?
Regards
Krzysztof

**Comment by harman-f** [comment]
Let us start with your log file.

**Reply by KJW13** [reply]
[MU1440-JTB-07411.12.171190067A.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11301005/MU1440-JTB-07411.12.171190067A.log)

**Comment by KJW13** [comment]
Regards
Krzysztof

**Comment by harman-f** [comment]
```
search File MU1440-ifs-root-part2-0x00ba0000-1C06F300.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f30628 (15926824)
End addr:   0x01ad0628
*** erasing:     0x00BA0000 .. 0x01AD0627 [01ADFFFF] .......................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AD0627            .........................................................................................
```

As visible in log file your flash process was interrupted before it finished.

RCC is now in a "bad" state. Quite unpredictable, depending on the missing files.
If connected via MMX, can you access /net/rcc/dev/fs0 ?

If not, you can use method to mount it:
https://mibwiki.one/doc/mount-rcc-fs0-6zJlPOu7Ml
Afterwards you can manually flash RCC.

Of course, as you only have 60s all this has to be quick ;)
You will have to change flashit parameters to ignore delete and verify.
To make flashing of ifs-root-stage2 possible in that short time frame.

**Reply by KJW13** [reply]
Thank you for your help.
The file fs0 is indeed not in the given path.
Unfortunately, I don't know how to perform the process you described in one go in 60s including restoring the fs0 files beforehand.
He doesn't want to make the problem worse by not following the instructions properly. Is there any guide to do it properly? If not, can you write me step by step how to do it?
Regards
Krzysztof

**Reply by harman-f** [reply]
as long as only ifs-root-stage 2 is written, there is no chance to make it worse ;)

1. Mount fs0 as mentioned on mibwiki
2. prepare binaries flashunlock/flashit/flashlock on SD card, as the once on unit are part of ifs-stage2 and most likely not working
3. flash ifs-root-stage2 (can be the patched one) directly to unit:

```
touch net/mmx/var/sm_disable_wd

on -f rcc /usr/bin/stop_efs_driver.sh
on -f rcc /sbin/devf-generic -s0x08000000,64M,,,128k,2,1 -r -D -P 1

mount -uw /net/mmx/fs/sda0
cd /net/mmx/fs/sda0
on -f rcc /net/mmx/fs/sda0/flashunlock
on -f rcc /net/mmx/fs/sda0/flashit -a 0x00ba0000 -d -f /net/mmx/fs/sda0/MU1440-ifs-root-part2-0x00ba0000-1C06F300.ifs
on -f rcc /net/mmx/fs/sda0/flashlock
```

**Reply by KJW13** [reply]
Thanks again for your help

The process didn't complete properly, but the device is working. I haven't checked all its features yet. Could the missing file (stop_efs_driver.sh) during resuscitation cause a problem?

Regards
Krzysztof
![Przechwytywanie](https://user-images.githubusercontent.com/131512521/234560811-29933074-8af2-4fb5-88bc-5df4fc00f7d7.PNG)

**Comment by harman-f** [comment]
Should all be finde now ;)
Great, that this worked

**Reply by Artem-V-Ozornin** [reply]
Hello, All and thank you very much @harman-f  for detailed instructions you shared above!
I've got the issue and situation ver similar to above, and proceed the steps from instruction below, but unfortunatelly it doesn't help :)
- the device still freezing and constantly rebooting
- no RCC connection by telnet
- with MMX connection no signs of /mnt/efs-persist directory exists 

I've attached my initial patch applying logs and screenshot of reflash both.
If you can advice on next steps, I would appreciate it much.
Thank you!
![Telnet screen - 2](https://github.com/user-attachments/assets/af3d9449-6451-47ee-b259-1f8caeaa622e)
[MU1352-JTB-07403.01.19116902D1.log](https://github.com/user-attachments/files/18815890/MU1352-JTB-07403.01.19116902D1.log)

**Comment by ferditayfurr** [comment]
Hello everyone

I'm not very good at reprogramming, but I'm not losing faith.

After trying to update MHI2_ER_VWG11_K3342_MU1427 to MHI2_ER_VWG13, my MHI2 froze after incorrectly removing the SD card. Now I get a Boot Loop error with the VW logo.
I have an RCC connection via PUTTY, but I can't type anything when it starts. Here's a screenshot of the error. After several attempts, I also lost my backup. I don't know how to do it. PUTTY won't let me use it, even for the help menu. Is there a code that does the whole thing so that the Green Menu
M.I.B._More-Incredible-Bash
is installed on the menu, and then I can unupdate to MHI2_ER_VWG13. Thank you for your help and understanding, as I'm French. Thank you for Google Translate.
![IMG_20250627_170826_resized_20250627_061040414(1)](https://github.com/user-attachments/assets/1a3b2413-1b76-4495-93ef-f415f6732611)

**Reply by ferditayfurr** [reply]
Hello again.

I formatted the SD card to FAT32, then extracted the contents of
M.I.B._More-Incredible-Bash to the root of the SD card, then added the MHI2_ER_VWG11_K3342_MU1427_PATCH patch to PATCHES, but RCC mode doesn't load the SD card, so I can't continue.
